### PR TITLE
Use mypy in github actions

### PR DESCRIPTION
**Pull request by Kane610**

### CODE REVIEW

File: aiounifi/dpi.py [hunk 1]
```diff
@@ -71,7 +47,7 @@ def site_id(self) -> str:
         return self.raw["site_id"]
 
 
-class DPIRestrictionApps(DPIItems):
+class DPIRestrictionApps(APIItems):
```
Comment (hunterjm) [inline aiounifi/dpi.py:50]: I believe this is the culprit for https://github.com/home-assistant/core/issues/57845

`DPIItems` processes state differently than `APIItems`.

File: aiounifi/dpi.py [hunk 2]
```diff
@@ -134,14 +108,14 @@ def dpiapp_ids(self) -> List[str]:
...
+    def enabled(self) -> Union[DPIRestrictionApps, None, bool]:
         """Are all apps in group enabled."""
         return self.apps and all(
-            [self.apps[id].enabled for id in self.apps if id in self.dpiapp_ids]
+            [self.apps[id].enabled for id in self.apps if id in self.dpiapp_ids]  # type: ignore
         )
 
 
-class DPIRestrictionGroups(DPIItems):
+class DPIRestrictionGroups(APIItems):
```
Comment (hunterjm) [inline aiounifi/dpi.py:118]: Here too

File: aiounifi/dpi.py (hunk 1 above)
Comment (Kane610) [reply aiounifi/dpi.py:50]: Thanks! I will have a look tonight!

File: aiounifi/dpi.py (hunk 2 above)
Comment (Kane610) [reply aiounifi/dpi.py:118]: Double thanks @hunterjm !

**Pull request merged**